### PR TITLE
fix: toStyleString omit nullish values

### LINKED ISSUE #8933: Property with nullish value in inline style should be omitted
### Astro Info

```block
Astro                    v3.4.0
Node                     v18.18.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

In JSX, `<div style={{ backgroundColor: condition ? '#d6f9dd' : undefined }} />` will not produce any style. But is astro template, this will produce `style="background-color:undefined"`

### What's the expected result?

behave as convention in JSX

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-47zegu?file=src%2Fpages%2Findex.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by MarvinXu**
## Changes
fix #8933 . 

- What does this change?
- Be short and concise. Bullet points can help!
- Before/after screenshots can help as well.
- Don't forget a changeset! `pnpm exec changeset`

## Testing
I've tested this utility function locally with following inputs:
```javascript
obj = {fontSize: '10px', color: undefined} // -> 'font-size:10px'
obj = {fontSize: null, color: 'red'} // -> 'color:red'
```

## Docs
No need to change as this is for compat with JSX

### CODE REVIEW

File: packages/astro/src/runtime/server/render/util.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ const kebab = (k: string) =>
 	k.toLowerCase() === k ? k : k.replace(/[A-Z]/g, (match) => `-${match.toLowerCase()}`);
 const toStyleString = (obj: Record<string, any>) =>
 	Object.entries(obj)
+		.filter(([k, v]) => v !== undefined && v !== null)
```
Comment (Princesseuh) [inline packages/astro/src/runtime/server/render/util.ts:31]: I'm not familiar with JSX, does it ignore all falsy values or just nullish? (and is there a reason to accept falsy values?)

Comment (MarvinXu) [reply packages/astro/src/runtime/server/render/util.ts:31]: I can't find its source code, so I did a little more [tests](https://reactplayground.vercel.app/#N4IglgdgJgpgHgOgFYGcQC5wFsAOB7AJwBcACAQRxxIDMC8sSByBAegp2RTkYB0IxchUgCUYAQwDGpWvSYFxU3v0HESwEhPliiMYXjykAvjToNGWqQFoo9FhIA2YGBCJK+EvBBSkPLsZBgCEgBeEhsJAFcsZyIEAHMYIgBRexholwAhAE8ASSgACnN9VwBKd09vEjoDEI0tHT0DfN8ifwhAsohq2PloQPy+EhJRSVjNcR0UtJiBiCGhkakEAGUiAjApAFk8WAAaQfmICPt7fbn5xbH6mCn0onz2TqHOkpBdkHZOOAwQPgF8VSXEyycwKVzlLykdi1fIlEIAPjUB3kRAiBDms3mJAAPAALACMJG8WVSwWAwAOWJYLBoniIyzAAC8YOgSAAGSnzDz2QisxhMM5Yobc3nswVCkUEVlrCIwcVYyWs6hiewoOWc4V4HlSpgkMQAIwkjHl831kgA1nE6BFoABhLWi-kkLIwE54ADuAo1JEc7QAEjAwHFcUQ+QajZzDIZ4QG3diWAT4QdOoY+Gn2nAAaRYMrjlDKCBDEA).

It seems to have more complicate validation rules than just nullish or falsy values.
```jsx
<h1 style={{
      // falsy value `0` is preserved in fontSize
      // fontSize: 0
     // color omits any value that is not a `color`
      color: ' ',
      color: 0,
      color: true,
      color: false,
      color: ' abc',
      backgroundColor: '  yellow ',
      lineHeight: 'abc'
    }}>Hello</h1>
```

Comment (ematipico) [inline packages/astro/src/runtime/server/render/util.ts:31]: Can't we use `.filter(Boolean)`?

Comment (MarvinXu) [reply packages/astro/src/runtime/server/render/util.ts:31]: https://github.com/withastro/astro/pull/8940#discussion_r1376193061 See my test. `font-size: 0` is a valid value

**Comment by MarvinXu** [comment]
Making a complete validation like JSX did (https://github.com/withastro/astro/pull/8940#discussion_r1376193061) requires a lot more effort. So I think we can stick with a quick and simple fix to solve common use cases like:
```jsx
// image may be undefined
<div style={{ backgroundImage: image && `url(${image.src})`}}></div>
```

I updated my code to only accept non-empty string and number

**Review by bluwy — APPROVED** [review]

**Review by ematipico — APPROVED** [review]

**Review by natemoo-re — COMMENTED** [review]
Just for clarity, this is implementation-specific behavior that isn't covered in the JSX spec. So the more complex behavior is just what React chooses to do.

I definitely agree that we should filter out `undefined`, `null`, and `false` though. Trying to think of any cases where we'd want to support anything other than strings and numbers, though.

**Pull request merged**